### PR TITLE
Move Query model to search promotions

### PR DESCRIPTION
**Pull request by kaedroho**
This PR moves the Query/QueryDailyHits models to search promotions. We don't use these anywhere else.

This makes the search module entirely focused on interfacing with search backends and it no longer provides any user interface. This makes it possible for us to later move this module out to a third-party package (after the deprecation period).

### CODE REVIEW

**Comment by kaedroho** [comment]
test

**Comment by kaedroho** [comment]
@gasman this has now been rebased

File: wagtail/contrib/search_promotions/tests.py [hunk 1]
```diff
@@ -485,3 +492,92 @@ def test_post(self):
...
+
+
+class TestGarbageCollectManagementCommand(TestCase):
+    def test_garbage_collect_command(self):
+        nowdt = datetime.now()
+        old_hit_date = (nowdt - timedelta(days=14)).date()
+        recent_hit_date = (nowdt - timedelta(days=1)).date()
+
+        # Add 10 hits that are more than one week old ; the related queries and the daily hits
+        # should be deleted bu the search_garbage_collect command.
```
Comment (allcaps) [inline wagtail/contrib/search_promotions/tests.py:504]: bu -> by

File: wagtail/contrib/search_promotions/models.py [hunk 2]
```diff
@@ -1,7 +1,93 @@
...
+        return cls.objects.get_or_create(
+            query_string=normalise_query_string(query_string)
+        )[0]
+
+    @classmethod
+    def get_most_popular(cls, date_since=None):
+        # TODO: Implement date_since
+        return (
+            cls.objects.filter(daily_hits__isnull=False)
+            .annotate(_hits=models.Sum("daily_hits__hits"))
```
Comment (allcaps) [inline wagtail/contrib/search_promotions/models.py:62]: Is the underscore in `_hits` a convention? Never saw private vars in an annotation before... I like it :D

File: wagtail/contrib/search_promotions/tests.py [hunk 3]
```diff
@@ -485,3 +492,92 @@ def test_post(self):
...
+        query_ids_to_be_deleted = []
+        for i in range(10):
+            q = Query.get("Hello {}".format(i))
+            q.add_hit(date=old_hit_date)
+            query_ids_to_be_deleted.append(q.id)
+
+        # Add 10 hits that are less than one week old ; these ones should not be deleted.
```
Comment (allcaps) [inline wagtail/contrib/search_promotions/tests.py:511]: Nitpick. Same as other comment? `old ; these` -> `old. These`

**Review by allcaps — APPROVED** [review]
LGTM, added some minor comments.

**Comment by lb-** [comment]
Note to self. I may be able to help get this polished/rebased and merged. 

Feedback on above is minor and PR has been approved.

**Comment by lb-** [comment]
Rebased, updated warnings to be removed in 6.0 warnings, force pushing. Will check CI and if all good will add release notes / changelog and merge.

**Comment by lb-** [comment]
Manually tested a few scenarios and migrations appear to work great & existing search promotions available.

I have added release notes / upgrade considerations & fixed the incorrect test file import.

If CI is all good I will merge this in.

**Pull request merged**